pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -474,19 +477,18 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
                 // `maxAmountsIn` and the exact underlying amount needs to be returned to the sender.
                 _sendTokenOut(callParams.sender, underlyingToken, maxAmountIn - actualAmountIn, callParams.wethIsEth);
             }
-
-            tokenIn = address(underlyingToken);
         } else {
+            actualAmountIn = amountIn;
+
             if (callParams.isStaticCall == false) {
-                _takeTokenIn(callParams.sender, IERC20(token), amountIn, callParams.wethIsEth);
+                _takeTokenIn(callParams.sender, IERC20(token), actualAmountIn, callParams.wethIsEth);
             }
-
-            actualAmountIn = amountIn;
-            tokenIn = token;
         }
 
         if (actualAmountIn > maxAmountIn) {
-            revert IVaultErrors.AmountInAboveMax(IERC20(token), amountIn, maxAmountIn);
+            IERC20 tokenIn = needToWrap ? underlyingToken : IERC20(token);
```

Comment:
I would do this the other way around.
`underlyingToken` only gets assigned in one of the `if / else` branches, and we're checking `needToWrap` twice in the function. It's not a huge deal, but it's unnecessary complexity IMO. Also this required moving `underlyingToken` outside of the loop.

Instead, you can declare `tokenIn` outside the loop as it was before this PR (not as a return value, but as a local variable at the same level where `underlyingToken` is right now. `tokenIn` is used in both branches, and assigned accordingly to either `underlyingToken` in the `if` branch, or to `token` in the `else` branch. And finally, here you just use `tokenIn` without checking `needToWrap` anymore, since it's already assigned correctly